snaps/how-to/use-environment-variables.md

```diff
@@ -0,0 +1,68 @@
+---
+sidebar_position: 3
+---
+
+# Use environment variables
+
+You can use environment variables to compile your Snap.
+This is useful if you want to use different variables for testing your Snap locally and publishing
+your Snap.
+You can use environment variables [on the command line](#use-environment-variables-on-the-command-line)
+or [in a `.env` file](#use-environment-variables-in-a-env-file).
+
+## Use environment variables on the command line
+
+1. Specify environment variables on the command line.
+    For example:
+
+    ```bash
+    PUBLIC_KEY=abc123 SNAP_ENV=dev
+    ```
+
+2. Build your Snap using the [Snaps CLI](../reference/cli/subcommands.md):
+
+    ```bash
+    yarn mm-snap build
+    ```
+
+## Use environment variables in a `.env` file
+
+1. Specify environment variables in a `.env` file.
+    For example:
+    
+    ```text title=".env"
+    PUBLIC_KEY=abc123
+    SNAP_ENV=dev
+    ```
+
+2. Specify the environment variables in your Snap's
+    [configuration file](../learn/about-snaps/files.md#configuration-file) using the `environment` option.
+    For example:
+
+    ```javascript title="snap.config.js"
+    module.exports = {
+        environment: {
+            SNAP_ENV: process.env.SNAP_ENV,
```

Comment:
We don't load `.env` automatically so doing this still requires `dotenv`. This looks good other than that!

Reply:
Does the docs framework have support for code tabs or something? It would be nice if we can show the TypeScript config too.

Reply:
@Mrtenz Updated with `dotenv` and TypeScript example.